### PR TITLE
Adding encoding check function to the output plugin

### PR DESCRIPTION
**Pull request by sw47**
@Microsoft/omsagent-devs 
please review - Moved the encoding checks from input plugins to output plugin so it will happen for all data types.

### CODE REVIEW

**Comment by agup006** [comment]
@Microsoft/omsagent-devs

**Comment by robbiezhang** [comment]
Signoff

File: source/code/plugins/out_oms.rb [hunk 1]
```diff
@@ -40,16 +40,18 @@ def shutdown
...
-      start = Time.now
-      
-      # This method will raise on failure alerting the engine to retry sending this data
-      OMS::Common.start_request(req, http)
-      
-      ends = Time.now
-      time = ends - start
-      count = record.has_key?('DataItems') ? record['DataItems'].size : 1
-      @log.debug "Success sending #{tag} x #{count} in #{time.round(2)}s"
+        unless req.nil?
```
Comment (niroyb) [inline source/code/plugins/out_oms.rb]: Indentation looks off here.

**Comment by niroyb** [comment]
Signed off

**Pull request merged**